docs/user-guide/command-line-interface.md

```diff
@@ -233,6 +234,24 @@ getSomeText | eslint --stdin --fix-dry-run --format=json
 
 This flag can be useful for integrations (e.g. editor plugins) which need to autofix text from the command line without saving it to the filesystem.
 
+#### `--fix-type`
+
+This option allows you to specify the type of fixes to apply when using either `--fix` or `--fix-dry-run`. The three types of fixes are:
+
+1. `problem` - fix potential errors in the code
```

Comment:
Yeah, I also went back and forth on this, and believe that "problem" is the correct term. Some of the other terms we use:

* "error" is a severity level, so definitely can't use that
* We use "messages" in the output data structure as a generic catchall for everything, so we don't need to worry about that.
* As best I can tell, "problem" is only ever used in the default formatter output. My feeling is that this isn't too confusing, but if others feel it is, I'd suggest changing the term to "messages" to echo what is in the data structure.

Reply:
Should we change the stylish formatter? We could use "issue" there, for example, whereas "issue" doesn't work as well for the rule type (IMO).

(I'm aware "issue" also has meaning in GitHub, but I'm not worried about confusion between lint output and GitHub at this point.)

Reply:
I wouldn't want to change the formatter as part of this PR, as I think that's a breaking change (we never know who is parsing the output). I'm open to changing it, though as I said, I also think it's fine if we don't.

Reply:
`problem` feels better to me than `messages` because `messages` sounds all-encompassing - it's not immediately clear how `messages` differs from `suggestion` or `style`